src/Datastore/DatastoreSessionHandler.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use Exception;
+
+/**
+ * Custom session handler backed by Cloud Datastore.
+ *
+ * {@see http://php.net/manual/en/class.sessionhandlerinterface.php}
+ *
+ * It uses the $savePath for the custom namespace which isolates the session
+ * data from your application data.
+ *
+ * Class DatastoreSessionHandler
+ * @package Google\Cloud\Datastore
+ */
+class DatastoreSessionHandler implements \SessionHandlerInterface
+{
+    const KIND = 'Session';
+
+    const GC_LIMIT = 1000;
+
+    /* @var DatastoreClient */
+    private $datastore;
+
+    public function open($savePath, $sessionName)
+    {
+        $this->datastore = new DatastoreClient(
```

Comment:
DatastoreClient instance needs to come via constructor injection, for unit tests and also so users can control the configuration of the client.

I'm not sure what to do with the savePath in this situation. You can pass a namespaceId into the `$options` array on `Datastore::key()`, so using the savePath as a namespace is still possible.

Any thoughts, @dwsupplee?